LXStudio-IDE/src/main/java/titanicsend/model/TEPanelModel.java

```diff
@@ -112,4 +115,23 @@ public TEPanelModel(ArrayList<LXPoint> points, TEVertex v0, TEVertex v1, TEVerte
     this.v1 = v1;
     this.v2 = v2;
   }
+
+  // Checks if two panels touch along an edge (not just at a vertex)
+  public boolean touches(TEPanelModel other) {
+    for (TEEdgeModel eThis : new TEEdgeModel[]{this.e0, this.e1, this.e2}) {
+      for (TEEdgeModel eOther : new TEEdgeModel[]{other.e0, other.e1, other.e2}) {
+        if (eThis == eOther) return true;
+      }
+    }
+    return false;
+  }
+
+  // Returns set of panels that touch along an edge (not just at a vertex)
+  public Set<TEPanelModel> neighbors() {
```

Comment:
nice, this should be useful. At some point we may want to do some profiling and may want to precompute or cache things like this that may be on hot paths.